subsys/bluetooth/shell/bt.c

```diff
@@ -346,41 +346,68 @@ static const char *scan_response_type_txt(uint8_t type)
 	}
 }
 
-static void scan_recv(const struct bt_le_scan_recv_info *info,
-		      struct net_buf_simple *buf)
+bool passes_scan_filter(const struct bt_le_scan_recv_info *info, struct net_buf_simple *buf)
 {
-	char le_addr[BT_ADDR_LE_STR_LEN];
-	char name[NAME_LEN];
-	struct net_buf_simple buf_copy;
 
 	if (scan_filter.rssi_set && (scan_filter.rssi > info->rssi)) {
-		return;
+		return false;
 	}
 
-	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	if (scan_filter.pa_interval_set &&
+	    (scan_filter.pa_interval > BT_CONN_INTERVAL_TO_MS(info->interval))) {
+		return false;
+	}
 
-	if (scan_filter.addr_set && !is_substring(scan_filter.addr, le_addr)) {
-		return;
+	if (scan_filter.addr_set) {
+		char le_addr[BT_ADDR_LE_STR_LEN] = {0};
+		int err;
+
+		err = bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+		if (err != 0) {
+			shell_error(ctx_shell, "Failed to convert addr to string: %d", err);
+			return false;
+		}
+
+		if (!is_substring(scan_filter.addr, le_addr)) {
+			return false;
+		}
 	}
 
-	if (scan_verbose_output) {
+	if (scan_filter.name_set) {
+		struct net_buf_simple buf_copy;
```

Comment:
Unused variable. `buf_copy` is only written to, never read from. Did you mean to parse `buf_copy` instead of `buf`? If so, it's also possible to pass a `buf` by-value instead of as a pointer into this function.